modules/CNN_blocks.py

```diff
@@ -0,0 +1,242 @@
+    
+    See more torch.nn.Conv2d
+    """
+    return nn.Conv2d(in_channels, out_channels, kernel_size=1, stride=stride, padding=padding)
+
+def conv3x3(in_channels, out_channels, stride=1, padding=1):
+    """ Make convolution layer with kernel_size is 3 
+        
+    See more torch.nn.Conv2d
+    """
+    return nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=stride, padding=padding)
+
+class ResNetNormalBlock(nn.Module):
+    """ Create Normal ResNet Block 
+    
+    Create ResNet block with two convolutional layers with skip connection.
+    All convolutional layer have normalize with BatchNorm layer.
+    Activation layer could be optional.
+    
+    Parameters
+    ----------
+    in_channels: int
+        number of channels in the input tensor.
+    out_channels: int
+        number of channels in the output tensor.
+    downsample: bool, optional
+        make block with downsampling.
+        downsampling decrease H and W of tensor in 2 times
+    activation: str, optional
+        set activation function in ResNetBlock
+        should be 'relu', 'sigmoid' or 'swish'
+    block_type: str, optional
+        set ResNet block type
+        should be 'A', 'B', 'C' or 'D'
+    
+    Raises
+    ------
+    ValueError
+        If downsample have unaccaptable value.
+        If block_type have unaccaptable value.
+        
+    See Also
+    --------
+    ResNetBottleneckBlock
+
+    [1]_Bag of Tricks for Image Classification with Convolutional Neural Networks. Part 4.1.
+    https://arxiv.org/pdf/1812.01187.pdf
+    """
+    
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 downsample=False,
+                 activation='relu',
+                 block_type='A'
+                 ):
+        super().__init__()
+        
+        if block_type not in ['A', 'B', 'C', 'D']:
+            raise ValueError('block_type have unacceptable value')
+        if not isinstance(downsample, bool):
+            raise TypeError('Downsample have unacceptable type')
+            
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.use_downsample = downsample
+        self.block_type = block_type        
+        
+        if activation == 'relu':
+            self.activation = nn.ReLU()
+        elif activation == 'sigmoid':
+            self.activation = nn.Sigmoid()
+        elif activation == 'swish':
+            self.activation = nn.SiLU()
+        else:
+            raise ValueError('activation have unacceptable value')
+            
+        if self.use_downsample:
+            if self.block_type == 'D':
+                self.downsample_block = nn.Sequential(
+                    nn.AvgPool2d(kernel_size=2, stride=2),
+                    conv1x1(self.in_channels, self.out_channels, stride=1),
+                    nn.BatchNorm2d(self.out_channels))
+                self.conv1 = conv3x3(self.in_channels, self.out_channels, stride=2)
```

Comment:
`self.conv1` does not depend on the `self.block_type`, only on `self.use_downsample`.
Move it to an upper-level condition